agent/util-scripts/test-bin/mock-cmd

```diff
@@ -17,15 +16,15 @@ _prog = os.path.basename(sys.argv[0])
 # ensure we don't run programs where they were not supposed to run.
 if _prog not in ("pbench-tool-meister-client", "yum", "scp"):
     with open(f"{_prog}.file", "a+") as ofp:
-        args = " ".join(sys.argv)
         ofp.write(f"{args}\n")
 
+_sleepers = ["dcgm-exporter", "node_exporter", "pmcd"]
 _sleep = 0
 if _prog == "pbench-sysinfo-dump":
     tgtdir = Path(sys.argv[1])
     sysinfo = sys.argv[2]
     (tgtdir / "contents.lis").write_text(sysinfo)
-elif _prog == "node_exporter" or (_prog == "prometheus" and sys.argv[1] == "run"):
+elif _prog in _sleepers or (_prog == "prometheus" and sys.argv[1] == "run"):
```

Comment:
Ugh; but I suppose the special case of `argv[1]` for prometheus precludes including it in `_sleepers`. It's still ugly!

Reply:
This is why I wanted a separate "mock" for the persistent tools, but with folks wanting to share it more, we get this kind of morass.

Hopefully, the more we convert `bash` to Python3, the more this kind of construct will go away.

Reply:
Sorry - I didn't realize prometheus was a specilal case: I thought it *was* a persistent tool.

Reply:
No problem!

Reply:
If it provides better aesthetics, we can have `_prog in _sleepers` in a separate `elif` from the Prometheus special case check.

In my opinion, the `_sleep` variable isn't buying us anything -- that is, if you don't divide Prometheus from the other sleepers, you can get rid of `_sleep` and just invoke `time.sleep()` in the "then" clause.  (I can see why you might want the invocation to be separate, but until we get a case other than `9999`, it doesn't buy us anything.)

So:
```
_sleepers = ["dcgm-exporter", "node_exporter", "pmcd", "prometheus"]
if _prog == "pbench-sysinfo-dump":
    tgtdir = Path(sys.argv[1])
    sysinfo = sys.argv[2]
    (tgtdir / "contents.lis").write_text(sysinfo)
elif _prog == "prometheus" and sys.argv[1] != "run")
    pass
elif _prog in _sleepers:
    time.sleep(9999)

sys.exit(0)
```

Reply:
Track in issue #2237.